runtime.c

```diff
@@ -385,6 +392,38 @@ main (int argc, char *argv[])
         exit(0);
     }
 
+    if (arg && strcmp(arg,"appimage-portable-home")==0) {
+        char portable_home[2048];
+        int mkdir_ret;
+
+        sprintf(portable_home, "%s.home", appimage_path);
+        mkdir_ret = mkdir(portable_home, S_IRWXU);
+        if (mkdir_ret == 0)
+            printf("Portable home folder created at %s\n", portable_home);
+        else if (mkdir_ret == EEXIST)
+            printf("Portable home folder %s already exists\n", portable_home);
+        else
+            printf("Error creating portable home folder at %s: %s\n", portable_home, strerror(errno));
+
+        exit(0);
+    }
+
+    if (arg && strcmp(arg,"appimage-portable-config")==0) {
+        char portable_config[2048];
+        int mkdir_ret;
+
+        sprintf(portable_config, "%s.config", appimage_path);
+        mkdir_ret = mkdir(portable_config, S_IRWXU);
```

Comment:
A `switch`-`case` construct is probably better, see comment about `portable_home`.